code/modules/mob/living/basic/heretic/flesh_stalker.dm

```diff
@@ -0,0 +1,45 @@
+/// Durable ambush mob with an EMP ability
+/mob/living/basic/heretic_summon/stalker
+	name = "Flesh Stalker"
+	real_name = "Flesh Stalker"
+	desc = "An abomination cobbled together from varied remains. Its appearance changes slightly every time you blink."
+	icon_state = "stalker"
+	icon_living = "stalker"
+	maxHealth = 150
+	health = 150
+	melee_damage_lower = 15
+	melee_damage_upper = 20
+	sight = SEE_MOBS
+	ai_controller = /datum/ai_controller/basic_controller/stalker
+
+/mob/living/basic/heretic_summon/stalker/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/ai_target_timer)
+
+	var/datum/action/cooldown/spell/jaunt/ethereal_jaunt/ash/jaunt = new(src)
+	jaunt.Grant(src)
```

Comment:
i'm wondering if this couldn't be an associative list for better extensibility in case someone adds more actions, like

```dm
/// Keyed by the datum we grant, values are the associated blackboard key
var/static/list/actions = list(
 /datum/action/cooldown/spell/jaunt/ethereal_jaunt/ash = null,
 /datum/action/cooldown/spell/shapeshift/eldritch = BB_SHAPESHIFT_ACTION,
 ...
)

for(var/action in actions)
 var/datum/action/spell = new action(src)
 spell.Grant(src)
 var/blackboard_key = actions[action]
 if(isnull(blackboard_key)
  continue
 ai_controller?.set_blackboard_key(blackboard_key, spell)
```

Do you think it's worth it?